12-watchtower-API.md

```diff
@@ -2,247 +2,340 @@
+* [Quality of Service data](#quality-of-service-data)
+	* [`accountability`](#accountability)
+* [Transaction Locator and Encryption Key](#transaction-locator-and-encryption-key)
+* [Encryption Algorithms and Parameters](#encryption-algorithms-and-parameters)
+* [Payment Modes](#payment-modes)
+* [No compression of justice transaction](#no-compression-of-justice-transaction)
 
-  * [Transaction Locator and Encryption Key](#transaction-locator-and-encryption-key)
-  * [Encryption Algorithms and Parameters](#encryption-algorithms-and-parameters)
-  * [WatchTower API](#WatchTower-api)
-  * [Signed Receipt Fields](#signed-receipt-fields)
-  * [Payment Modes](#payment-modes)
-  * [Number of updates](#number-of-updates)
-  * [No compression of justice transaction](#no-compression-of-justice-transaction)
+## WatchTower discovery
+At this point we're leaving the client/server connection to be protocol agnostic. How the Lightning node finds the WatchTower or how the WatchTower announces their presence and services provided is not specified.
 
-## Connection establishment
-Connections between the client and the server can be long-lived or restarted for every single appointment.
+Therefore, we are assuming that the client and server are connected and that the client have learnt what Quality of Service (`qos`) the tower is offering. Moreover, we assume that the client has an authentication token `auth_token` to prove he's entitled to use the service if required.	
 
-		+-------+                      +-------+
-		|   A   |--(1)--- wt_init ---->|   B   |
-		|       |<-(2)--- wt_init -----|       |
-		+-------+                      +-------+
+We have not defined how a client can find a list of servers to hire yet. We assume the client has found a server and the server is offering a watching service. The service can either be the basic service or an accountable quality of service. To deal with pre-payments (when necessary), the client may have an authentication token that the server can verify when accepting the job (e.g. a blinded token).
+
+## WatchTower services
+
+### Basic Service
+The customer can hire the WatchTower to watch for breaches on the blockchain and relay a justice transaction on their behalf. The customer receives an acknowledgement when the WatchTower has accepted the job, but the hiring protocol does not guarantee the transaction inclusion.
+
+### Quality of Service
+Quality of Service (`qos`) builds on top of the basic service provided by a tower and it's optionally provided. Different kinds of QoS can be offered by the tower.
+
+For now we are defining a single type of `qos`: `accountability`.
+
+#### `accountability`
+
+A WatchTower provides a signed receipt to the customer. This is considered reputational accountability as the customer has publicly verifiable cryptographic evidence the WatchTower was hired. The receipt can be used to prove the WatchTower did not relay the justice transaction on their behalf and/or request a refund.
```

Comment:
should this be server/client? or watchtower? in the descriptions here .

(from here up to watchtower service client).

Reply:
I'll leave it like that for now at least then